### DISCUSSION TITLE
keyball61_右側のUSB-Cが反応しない。

### DISCUSSION THREAD

**Discussion by uesiman** [Yowkees/keyball#418] [Q&A]
購入させていただき、ほかの方のエラーを拝見しながらなんとか組み立てたのですが、原因がわからずお伺いさせてください。
〇事象概要
keyball_61（右手マウスボール）で右側のUSB-C口でPCで差してもキーボードが認識しない。

〇事象詳細
①右側のUSBーCでPCに接続しても両方の画面の液晶には何も表示されない。また、USBーCの赤いのランプも両側点灯しない。
②左側のUSBーCで接続するとキーボートとして認識されて、ファームウェアの書き込みもできる。
③②の際には左側の液晶には数字が表示されて、右側の液晶には「KeyBall」という表示がされる。また、右、左USB-Cの接続の赤いランプが点灯している。

以下は問題となっている右側裏面の基盤の状況です。対処方法をご教示いただけますと幸いです。

![image_67212289](https://github.com/Yowkees/keyball/assets/150325717/300f69bc-7725-4976-9c89-ab404c757bd9)
![image_67220225](https://github.com/Yowkees/keyball/assets/150325717/cc25f666-0d1c-4cb7-bf89-e3c0e315b86f)
![image_67225857](https://github.com/Yowkees/keyball/assets/150325717/4c0f4e53-6d03-490e-b8e2-3de5f1a5e9e7)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

現象、了解しました。
確認したいのですが、左手側にUSBケーブルを接続した場合には、すべて正常に動作するということで良かったでしょうか？（トラックボール、キー入力、OLED）

順番に問題を切り分けたいと思いますので、右手側のProMicroをKeyballから取り外して、ProMicro単体ではファームの書き込みができるかご確認ください。ProMicro単体でリセットをかけるには"GND"と"RST"の端子をピンセットでショートします。

以上、お手数ですがご確認お願いいたします。

**Reply by uesiman** [reply]
ご返信遅くなり申し訳ありません。


> 左手側にUSBケーブルを接続した場合には、すべて正常に動作するということで良かったでしょうか？（トラックボール、キー入力、OLED）

はい、全て正常に作動します。

>promicro単体では書き込みできるか？

単体では書き込みすることができました。

**Comment by Yowkees** [comment]
ご確認ありがとうございます。

症状から考えると、右手側のProMicro周辺の半田付けで、電源ライン(RAW,VCC)が他の線とショートしているのではないかと思われます。
光の加減かもしれませんが、下記画像で丸をつけた個所は隣とショートしていないでしょうか？
また、半田がキーソケット内に流れ込んで隣とショートする可能性もあり、テスターで隣とショートしている端子がないかご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。
![KB61_rightfix](https://github.com/Yowkees/keyball/assets/84209772/fcefc120-6bdd-4491-bcb8-df53c52fb702)
